Watchman.Discord/Areas/Administration/Controllers/AdministrationController.cs

```diff
@@ -38,17 +39,12 @@ public AdministrationController(IQueryBus queryBus, UsersService usersService, D
         }
 
         [AdminCommand]
-        [DiscordCommand("messages")]
-        public async Task ReadUserMessages(DiscordRequest request, Contexts contexts)
+        public async Task ReadUserMessages(MessagesCommand command, Contexts contexts)
         {
-            var mention = request.GetMention();
-            var selectedUser = await this._usersService.GetUserByMentionAsync(contexts.Server, mention);
-            if (selectedUser == null)
-            {
-                throw new UserNotFoundException(mention);
-            }
+            var selectedUser = await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            var messagesService = this._messagesServiceFactory.Create(contexts);
+            var timeRange = new TimeRange(DateTime.UtcNow - command.Time, DateTime.UtcNow);
```

Comment:
jak dobrze pamiętam to TimeRange ma wbudowane funkcje FromNow i ToNow
możesz użyć tutaj ToNow